tip-router-operator-cli/src/lib.rs

```diff
@@ -113,53 +132,128 @@ fn emit_inconsistent_tree_node_amount_dp(
+                    (protocol_fee_bps as u128)) /
+                    10000u128) as u64;
+                let validator_fee_amount = (((tip_distribution_meta.total_tips as u128) *
+                    (tip_distribution_meta.validator_fee_bps as u128)) /
+                    10000u128) as u64;
+                let remaining_tips =
+                    tip_distribution_meta.total_tips - protocol_fee_amount - validator_fee_amount;
+
+                // Create nodes in order: protocol fee, validator fee, delegators
+                let mut tree_nodes = Vec::with_capacity(2 + stake_meta.delegations.len());
+
+                // 1. Protocol Fee Node
+                let (protocol_fee_recipient, _) = Pubkey::find_program_address(
+                    &[b"protocol_fee", &(0u64).to_le_bytes()],
+                    &stake_meta_coll.tip_distribution_program_id
+                );
 
-                let tip_distribution_meta = stake_meta.maybe_tip_distribution_meta.unwrap();
+                let (claim_status_pubkey, claim_status_bump) = Pubkey::find_program_address(
+                    &[
+                        ClaimStatus::SEED,
+                        &protocol_fee_recipient.to_bytes(),
+                        &tip_distribution_meta.tip_distribution_pubkey.to_bytes(),
+                    ],
+                    &JitoTipDistribution::id()
+                );
+
+                tree_nodes.push(TreeNode {
+                    claimant: protocol_fee_recipient,
+                    claim_status_pubkey,
+                    claim_status_bump,
+                    staker_pubkey: Pubkey::default(),
+                    withdrawer_pubkey: Pubkey::default(),
+                    amount: protocol_fee_amount,
+                    proof: None,
+                });
+
+                // 2. Validator Fee Node
+                let (claim_status_pubkey, claim_status_bump) = Pubkey::find_program_address(
+                    &[
+                        ClaimStatus::SEED,
+                        &stake_meta.validator_node_pubkey.to_bytes(),
+                        &tip_distribution_meta.tip_distribution_pubkey.to_bytes(),
+                    ],
+                    &JitoTipDistribution::id()
+                );
+
+                tree_nodes.push(TreeNode {
+                    claimant: stake_meta.validator_node_pubkey,
+                    claim_status_pubkey,
+                    claim_status_bump,
+                    staker_pubkey: Pubkey::default(),
+                    withdrawer_pubkey: Pubkey::default(),
+                    amount: validator_fee_amount,
+                    proof: None,
+                });
+
+                // 3. Delegator Nodes
+                for delegation in &stake_meta.delegations {
+                    let delegator_share = (((remaining_tips as u128) *
+                        (delegation.lamports_delegated as u128)) /
+                        (stake_meta.total_delegated as u128)) as u64;
```

Comment:
when casting to u64 from a larger number type, it's better to use something like `u64::try_from`. That way it will handle any errors if it over flows. However, if you intend to lose the precision, then it's fine to just use `u64::from` or regular casting. Worth noting this as it's something auditors will often point out.